service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -46,10 +74,84 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             caseDetails.getData().put("hiddenOrders", hiddenOrders);
         }
 
+        if ("FPLA-2710".equals(migrationId)) {
+            CaseData caseData = getCaseData(caseDetails);
+            if (casesWithUploadDraftCMOs.contains(caseData.getId()) && isNotEmpty(caseData.getDraftUploadedCMOs())) {
+                log.info(
+                    "Migrating draft CMOs to Hearing orders draft bundles - case reference {} Number of Draft CMOs {}",
+                    caseData.getId(), caseData.getDraftUploadedCMOs().size());
+
+                List<Element<HearingOrdersBundle>> migratedBundles = migrateDraftCMOsToHearingBundles(caseData);
+                caseDetails.getData().put("hearingOrdersBundlesDrafts", migratedBundles);
+                caseDetails.getData().put("draftUploadedCMOs", caseData.getDraftUploadedCMOs());
+
+                log.info(
+                    "Completed migration to Hearing orders draft bundles. case reference {} Number of bundles {}",
+                    caseData.getId(), migratedBundles.size());
+            }
+        }
+
         caseDetails.getData().remove(MIGRATION_ID_KEY);
         return respond(caseDetails);
     }
 
+    private List<Element<HearingOrdersBundle>> migrateDraftCMOsToHearingBundles(CaseData caseData) {
+
+        List<Element<HearingOrder>> draftUploadedCMOs = caseData.getDraftUploadedCMOs();
+
+        List<Element<HearingOrdersBundle>> hearingOrdersBundles = defaultIfNull(
+            caseData.getHearingOrdersBundlesDrafts(), new ArrayList<>());
+
+        List<UUID> removedDraftCMOs = new ArrayList<>();
+
+        removeDraftCMOsFromHearingBundles(hearingOrdersBundles, draftUploadedCMOs);
+
+        draftUploadedCMOs.forEach(draftElement -> {
+            Optional<Element<HearingBooking>> hearingOptional = caseData.getHearingLinkedToCMO(draftElement.getId());
+
+            // if hearing exists linked with draft CMO
+            hearingOptional.ifPresent(hearingElement -> {
+                String title = draftElement.getValue().getType() == AGREED_CMO ? "Agreed CMO discussed at hearing"
+                    : "Draft CMO from advocates' meeting";
+
+                draftElement.getValue().toBuilder()
```

Comment:
I don't think this is updating the title -> tobuilder creates another instance that is not reassigned

Reply:
This code is replaced with `DraftOrderService. migrateCmoDraftToOrdersBundles`